From 323e03c7c07e64584fdd41b7c41b09ed080dab9f Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Sat, 4 Nov 2023 11:21:16 +0800
Subject: [PATCH 17/17] semaphores: temporary hack for sem_timedwait

---
 deps/v8/src/base/platform/semaphore.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git deps/v8/src/base/platform/semaphore.cc deps/v8/src/base/platform/semaphore.cc
index 80fe55b77c3..1b4feac2240 100644
--- deps/v8/src/base/platform/semaphore.cc
+++ deps/v8/src/base/platform/semaphore.cc
@@ -90,7 +90,11 @@ bool Semaphore::WaitFor(const TimeDelta& rel_time) {
 
   // Wait for semaphore signalled or timeout.
   while (true) {
+#if V8_OS_MACOSX
+    int result = sem_wait(&native_handle_); // This is a dirty hack to make it compile; it is not expected to work.
+#else
     int result = sem_timedwait(&native_handle_, &ts);
+#endif
     if (result == 0) return true;  // Semaphore was signalled.
 #if V8_LIBC_GLIBC && !V8_GLIBC_PREREQ(2, 4)
     if (result > 0) {
