--- plugins/history2/history2logger.h
+++ plugins/history2/history2logger.h	2024-07-17 23:19:16.000000000 +0800
@@ -94,7 +94,7 @@
 	 * from Kopete::Contact @param c in the given @param sens
 	 */
 	QList<Kopete::Message> readMessages(int lines,
-	                                    int offset=0, const Kopete::MetaContact *c=false, bool reverseOrder=true);
+	                                    int offset=0, const Kopete::MetaContact *c=0, bool reverseOrder=true);
 
 	/**
 	 * Same as the following, but for one date. I did'nt reuse the above function
