--- a/cmake/modules/FindMySQLAmarok.cmake	2015-06-05 12:18:05.000000000 +0800
+++ b/cmake/modules/FindMySQLAmarok.cmake	2024-07-19 00:35:13.000000000 +0800
@@ -13,10 +13,10 @@
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
 if(NOT WIN32)
-    find_program(MYSQLCONFIG_EXECUTABLE NAMES mysql_config mysql_config5 HINTS ${BIN_INSTALL_DIR})
+    set(MYSQLCONFIG_EXECUTABLE @PREFIX@/lib/@MYSQL@/bin/mysql_config)
 endif(NOT WIN32)
 
-find_path(MYSQL_INCLUDE_DIR mysql.h PATH_SUFFIXES mysql mysql5/mysql)
+set(MYSQL_INCLUDE_DIR @PREFIX@/include/@MYSQL@/mysql)
 
 if(MYSQLCONFIG_EXECUTABLE)
     exec_program(${MYSQLCONFIG_EXECUTABLE} ARGS --cflags RETURN_VALUE _return_VALUE OUTPUT_VARIABLE MYSQL_CFLAGS)
