--- src/debug.h.orig	2022-08-01 02:48:42.000000000 +0800
+++ src/debug.h	2022-11-07 14:23:42.000000000 +0800
@@ -23,16 +23,21 @@
 extern char *WORKQUEUE_DEBUG_IDENT;
 
 #if defined(__linux__)
-
-#include <linux/unistd.h>
-#include <sys/syscall.h>
-#include <unistd.h>
-
+# include <linux/unistd.h>
+# include <sys/syscall.h>
+# include <unistd.h>
 # define THREAD_ID ((pid_t)  syscall(__NR_gettid))
 #elif defined(__sun)
 # define THREAD_ID (pthread_self())
 #elif defined(__FreeBSD__)  /* FIXME -- could use thr_self() */
 # define THREAD_ID (0)
+#elif defined(__APPLE__)
+# include <AvailabilityMacros.h>
+# if (MAC_OS_X_VERSION_MAX_ALLOWED < 1060 || defined(__POWERPC__))
+# define THREAD_ID (pthread_mach_thread_np(pthread_self()))
+# else
+# define THREAD_ID (pthread_threadid_np())
+# endif /* __APPLE__ */
 #elif defined(_WIN32)
 # define THREAD_ID (int)(GetCurrentThreadId())
 #else 
