--- kernel/power/KERNEL.PPC970.orig	2024-12-30 20:49:18.000000000 +0800
+++ kernel/power/KERNEL.PPC970	2024-12-30 20:49:30.000000000 +0800
@@ -83,6 +83,10 @@
 CTRSMKERNEL_RN	=  ztrsm_kernel_LT.S
 CTRSMKERNEL_RT	=  ztrsm_kernel_RT.S
 
+SSCALKERNEL  = ../arm/scal.c
+DSCALKERNEL  = ../arm/scal.c
+CSCALKERNEL  = ../arm/zscal.c
+ZSCALKERNEL  = ../arm/zscal.c
 
 SROTKERNEL   = ../arm/rot.c
 DROTKERNEL   = ../arm/rot.c
