--- include/pthread_workqueue.h.orig	2022-08-01 02:48:42.000000000 +0800
+++ include/pthread_workqueue.h	2022-11-07 18:04:49.000000000 +0800
@@ -58,7 +58,7 @@
 #if defined(__cplusplus)
 	extern "C" {
 #endif
-		
+
 int _PWQ_EXPORT pthread_workqueue_create_np(pthread_workqueue_t * workqp,
                const pthread_workqueue_attr_t * attr);
 
@@ -90,12 +90,14 @@
 int _PWQ_EXPORT pthread_workqueue_init_np(void);
 
 /* NOTE: these are not part of the Darwin API */
+#ifndef __APPLE__
 unsigned long _PWQ_EXPORT pthread_workqueue_peek_np(const char *);
 void _PWQ_EXPORT pthread_workqueue_suspend_np(void);
 void _PWQ_EXPORT pthread_workqueue_resume_np(void);
 void _PWQ_EXPORT pthread_workqueue_signal_np(void);
 
 void _PWQ_EXPORT (*libpwq_thread_cleanup_handler)();
+#endif
 #if defined(__cplusplus)
 	}
 #endif
