--- boost/smart_ptr/detail/spinlock_gcc_atomic.hpp.orig	2021-04-14 00:22:55.000000000 +0800
+++ boost/smart_ptr/detail/spinlock_gcc_atomic.hpp	2022-11-27 19:19:39.000000000 +0800
@@ -30,7 +30,15 @@
 {
 public:
 
-    unsigned char v_;
+    // `bool` alignment is required for Apple ppc32
+    // https://github.com/boostorg/smart_ptr/issues/105
+    // https://github.com/PurpleI2P/i2pd/issues/1726
+    // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=107590
+    union
+    {
+        unsigned char v_;
+        bool align_;
+    };
 
 public:
 
@@ -80,6 +88,6 @@
 } // namespace detail
 } // namespace boost
 
-#define BOOST_DETAIL_SPINLOCK_INIT {0}
+#define BOOST_DETAIL_SPINLOCK_INIT {{0}}
 
 #endif // #ifndef BOOST_SMART_PTR_DETAIL_SPINLOCK_GCC_ATOMIC_HPP_INCLUDED
