--- SystemDirectory.xs.orig	2022-02-21 13:33:28.000000000 +0800
+++ SystemDirectory.xs	2024-06-22 03:47:36.000000000 +0800
@@ -63,7 +63,7 @@
 #endif
     const_uv(NSAllApplicationsDirectory);
     const_uv(NSAllLibrariesDirectory);
-#if (defined(MAC_OS_X_VERSION_10_6) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_6))
+#if (defined(MAC_OS_X_VERSION_10_6) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_6) && !defined(__ppc__))
     const_uv(NSInputMethodsDirectory);
     const_uv(NSMoviesDirectory);
     const_uv(NSMusicDirectory);
@@ -114,7 +114,7 @@
 #endif
         case NSAllApplicationsDirectory:
         case NSAllLibrariesDirectory:
-#if (defined(MAC_OS_X_VERSION_10_6) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_6))
+#if (defined(MAC_OS_X_VERSION_10_6) && (MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_6) && !defined(__ppc__))
         case NSInputMethodsDirectory:
         case NSMoviesDirectory:
         case NSMusicDirectory:
